### PR TITLE
Add a GHA to run unit tests

### PR DESCRIPTION
**Pull request by surajkumar**
### What changed

Added a new GHA

### Why did it change
So we can run unit tests via a GHA

### Issue tracking

- [OJ-1470](https://govukverify.atlassian.net/browse/OJ-1470)

### Environment variables or secrets

- [X] No environment variables or secrets were added or changed

[OJ-1470]: https://govukverify.atlassian.net/browse/OJ-1470?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+      - name: Install Dependencies
+        run: npm install
+      - name: Run Tests
+        run: npm run compile && npm run unit
```
Comment (scronk1) [inline .github/workflows/unit-tests.yml]: Very minor but `npm run test` does this exact thing if you wanted to be more brief

**Review by scronk1 — APPROVED** [review]

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (surajkumar) [reply .github/workflows/unit-tests.yml]: Cheers, I actually copied it from the package.json but will do this suggestion

Comment (merlinc) [reply .github/workflows/unit-tests.yml]: That was going to be my suggestion too :)

**Review by scronk1 — CHANGES_REQUESTED** [review]
...

File: .github/workflows/unit-tests.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+      - name: Install Dependencies
+        run: npm install
+      - name: Run lint - prettier
+        run: npm run lint:prettier
```
Comment (scronk1) [inline .github/workflows/unit-tests.yml]: Any reason you're doing `npm run lint:prettier` and not just `npm run lint`

Comment (surajkumar) [reply .github/workflows/unit-tests.yml]: Copied it from common - this is a good spot, updated

**Review by scronk1 — APPROVED** [review]

**Pull request merged**